pylint/checkers/unicode.py

```diff
@@ -0,0 +1,728 @@
+            self._check_name("const", name, node)
+
+    @utils.check_messages("non-ascii-identifier")
+    def visit_assignname(self, node: nodes.AssignName) -> None:
+        """check module level assigned names"""
+        frame = node.frame()
+        assign_type = node.assign_type()
+        if isinstance(assign_type, nodes.Comprehension):
+            # original inlinevar
+            self._check_name("variable", node.name, node)
+        elif isinstance(frame, nodes.Module):
+            # Here a lot of checks were done in the original module,
+            # but we call it a day and say everything is variable
+            self._check_name("variable", node.name, node)
+        elif isinstance(frame, nodes.FunctionDef):
+            if not hasattr(node, "_is_ascii_only"):
+                # only check if not already done
+                self._check_name("variable", node.name, node)
+        elif isinstance(frame, nodes.ClassDef):
+            self._check_name("attr", node.name, node)
+        else:
+            # Just to make sure we check EVERYTHING (!)
+            self._check_name("variable", node.name, node)
+
+    @utils.check_messages("non-ascii-identifier", "non-ascii-name")
+    def visit_classdef(self, node: nodes.ClassDef) -> None:
+        self._check_name("class", node.name, node)
+        for attr, anodes in node.instance_attrs.items():
+            if not any(node.instance_attr_ancestors(attr)):
+                self._check_name("attr", attr, anodes[0])
+
+    @staticmethod
+    def _get_module_names(module_name: str, alias: Optional[str] = None) -> List[str]:
+        result = module_name.split(".")
+        if alias is not None:
+            result.append(alias)
+        return result
+
+    def _check_module_import(
+        self, node: Union[nodes.ImportFrom, nodes.Import], is_import_from: bool = False
+    ):
+        names = []
+        for module_name, alias in node.names:
+            names += self._get_module_names(module_name, alias)
+        if not (is_import_from and names == ["*"]):
+            # Ignore ``from xyz import *``
+            self._check_name("module", names, node)
+
+    @utils.check_messages("non-ascii-identifier", "non-ascii-name")
+    def visit_import(self, node: nodes.Import) -> None:
+        self._check_module_import(node)
+
+    @utils.check_messages("non-ascii-identifier", "non-ascii-name")
+    def visit_importfrom(self, node: nodes.ImportFrom) -> None:
+        self._check_name("module", self._get_module_names(node.modname), node)
+        self._check_module_import(node, is_import_from=True)
+
+    @utils.check_messages("non-ascii-identifier", "non-ascii-name")
+    def visit_call(self, node: nodes.Call) -> None:
+        # lets check if the used keyword args are correct
+        for keyword in node.keywords:
+            self._check_name("argument", keyword.arg, keyword)
+
+
+def register(linter: PyLinter) -> None:
+    """Required method to auto register this checker"""
```

Comment:
This comment was forgotten because it was the 21st hidden comments on the previous report @CarliJoy 😄 (and only the first 20 get shown when clicking to display hidden comments)

Reply:
@Pierre-Sassoulas  no this comment was forgotten because for some strange reason I could not add it to the batch/commiot it.
I wanted to to it manually but forgot about it...
Did it now